time_manager/time_manager.F90

```diff
@@ -1213,9 +1213,27 @@ function real_to_time_type(x,err_msg) result(t)
   real :: tps
   real :: a
   tps = real(ticks_per_second)
-  a = x/spd
+
+  select type (x)
+  type is (real(kind=r4_kind))
+    a = x/spd
+  type is (real(kind=r8_kind))
+    a = x/spd
+  class default
+    call error_mesg('time_manager_mod::real_to_time_type', 'unsupported kind', FATAL)
+  end select
+
   days = safe_rtoi(a,do_floor)
-  a = x - real(days)*spd
+
+  select type (x)
+  type is (real(kind=r4_kind))
+    a = x - real(days)*spd
+  type is (real(kind=r8_kind))
+    a = x - real(days)*spd
```

Comment:
Should these casts be done on all on `x` also?

Reply:
The kind of `a`, `spd` and `real(days)` will depend on how FMS is compiled. Is it problematic if it does not match the kind of `x`?

Reply:
If you do
```fortran
  type is (real(kind=r4_kind))
    a = real(x) - real(days)*spd
  type is (real(kind=r8_kind))
    a = real(x) - real(days)*spd
```
then it should compile without warnings.  I tried to compile:
```fortran
program test
 real(kind=4) :: r4
 real(kind=8) :: r8 = 10.0d0
 real(kind=8) :: r = 5.0d0
   r4 = real(r8)/real(r)
end program test
```
with `--warn-pedantic`, and it gave no errors.